pkg/rules/engine/engine.go

```diff
@@ -307,3 +307,77 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+// GetFilters returns a mediated array of all signature's filters
+func (engine *Engine) GetFilters() ([]protocol.Filter, error) {
+	allFilters := make([]protocol.Filter, 0)
+	engine.signaturesMutex.RLock()
+	defer engine.signaturesMutex.RUnlock()
+	for sig := range engine.signatures {
+		metadata, _ := sig.GetMetadata()
+		selectors, err := sig.GetSelectedEvents()
+		if err != nil {
+			return []protocol.Filter{}, fmt.Errorf("error getting filters: missing selectors for base filters from signature %s", metadata.Name)
+		}
+		baseFilters := []protocol.Filter{}
+		for _, s := range selectors {
+			baseFilters = append(baseFilters, s.ToFilters()...)
+		}
+		if len(baseFilters) == 0 {
+			return []protocol.Filter{}, fmt.Errorf("error getting filters: missing base filters from signature %s", metadata.Name)
+		}
+		allFilters = append(allFilters, baseFilters...)
+		filters, err := sig.GetFilters()
+		if err != nil {
+			return []protocol.Filter{}, fmt.Errorf("error getting filters: %v", err)
+		}
+		allFilters = append(allFilters, filters...)
+	}
+	res := mediateFilters(allFilters)
+	return res, nil
+}
+
+// mediateFilters performs mediation logics on a list of filters
+func mediateFilters(filters []protocol.Filter) []protocol.Filter {
+	type filterHeader struct {
+		field    string
+		operator protocol.FilterOperator
+	}
+	// remove := func(s []interface{}, i int) []interface{} {
+	// 	s[i] = s[len(s)-1]
+	// 	return s[:len(s)-1]
+	// }
+	filterGroups := make(map[filterHeader]*protocol.Filter)
+	// For now - mediation should be like a SQL GROUP BY clause, with no value duplications
```

Comment:
So we're not mediating among different operators ? I think "it is fine" if we consider the current filtering mechanism also does not mediate that (giving mutual exclusive operators would be problematic with what we currently have now as well). I believe we will have to handle that eventually ? (likely outside the scope of this POC ?).

Reply:
So the filters in tracee-ebpf sort of already mediate between their operators. I have kept that logic in my filters refactor (#1750). I also believe we will face that problem if and when this filter system gets more complex, but I don't want to get ahead of myself with that.

Reply:
> but I don't want to get ahead of myself with that.

agreed, we should have it simple and working for most cases now.